internal/ethapi/api.go

```diff
@@ -709,9 +722,9 @@ func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, args CallArgs) (h
 
 	// Create a helper to check if a gas allowance results in an executable transaction
 	executable := func(gas uint64) bool {
-		args.Gas = hexutil.Uint64(gas)
+		args.Gas = (*hexutil.Uint64)(&gas)
```

Comment:
What's with this construct? Why cast and cast back?

Reply:
We're casting a `uint64` to a `*hexutil.Uint64`, because `args.Gas` is now a pointer.